include/seqan3/alignment/pairwise/alignment_range.hpp

```diff
@@ -18,133 +18,39 @@
 namespace seqan3
 {
 
-/*!\brief The alignment
+/*!\brief An input range over the alignment results generated by the underlying alignment executor.
  * \ingroup pairwise_alignment
- * \tparam alignment_executor_type The buffer type for the alignment stream.
+ * \implements std::ranges::input_range
+ *
+ * \tparam alignment_executor_type The type of the underlying alignment executor.
  *
  * \details
- * \attention This class considers moveable-only ranges.
  *
- * Provides a stream-like range interface over the alignments instances that are computed in a
- * seqan3::detail::alignment_executor_two_way executor.
+ * Provides a lazy input-range interface over the alignment results generated by the underlying alignment
+ * executor. The alignments are computed in a lazy fashion, such that when incrementing the next alignment
+ * result is fetched from the executor. The alignment result will be cached such that dereferencing the
+ * associated iterator is constant.
  */
 template <typename alignment_executor_type>
-//TODO requires alignment_executor_concept<alignment_executor_type>
```

Comment:
Is there no concept?

Reply:
no. Not yet. It wasn't clear yet what the concept would really be about. At the moment this is very strongly coupled. In case more generic use cases are developed a concept can be created. But for now, this is not neccessary.

Reply:
If it is strongly coupled, why is this a generic type?